### PR TITLE
Fix #39 : Added Shimmer Loading effect in RecyclerView

### LINKED ISSUE #39: Feature: Custom Loading dialog in HomeFragment while data gets fetched
### Feature description

Need to implement a custom dialog in HomeFragment which will be visible till the data is fetched from the Firebase.

Use Google's LottieAnimation as the view for the Dialog (as in AboutFragment).

### 👀 Have you spent some time to check if this issue has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by MohitGupta121**
Fixed #39  Shimmer Loading effect in HomeFragment while data gets fetched

Below is the demo video for the issue fixed:- 

https://user-images.githubusercontent.com/76530270/158065537-8558a79d-c9f4-40ee-a304-807034d7de84.mp4

### CODE REVIEW

**Comment by MohitGupta121** [comment]
@fin1te can you please review my this PR for issue #39

**Review by fin1te — COMMENTED** [review]
Had a small bug where visibility was not set to gone after data was loaded.

**Review by fin1te — COMMENTED** [review]
Fixed the minor issues within the last commit, overall everything was perfect.

Good work @MohitGupta121! :D

**Comment by MohitGupta121** [comment]
> Had a small bug where visibility was not set to gone after data was loaded.

No, I add the visibility Gone but let me see

File: app/src/main/java/com/finite/gdscphcet/ui/HomeFragment.kt [hunk 1]
```diff
@@ -129,9 +129,12 @@ class HomeFragment : Fragment() {
                         requireContext(),
                         upcomingEventsList.reversed() as MutableList<UpcomingEventModel>
                     )
+                    upcomingShimmer.stopShimmer()
                     upcomingShimmer.visibility = View.GONE
 
                 } else{
+                    upcomingShimmer.stopShimmer()
```
Comment (MohitGupta121) [inline app/src/main/java/com/finite/gdscphcet/ui/HomeFragment.kt:136]: Bro I not think so here is it needed as when data loaded and there is no upcoming event then it show placeholder for no upcoming event.

After that is it working fine ??

Comment (fin1te) [reply app/src/main/java/com/finite/gdscphcet/ui/HomeFragment.kt:136]: Yes it is working fine, checked after making the changes

**Pull request merged**